lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php

```diff
@@ -273,7 +273,13 @@ public function bindValue($param, $value, int $type = ParameterType::STRING) : v
      */
     public function bindParam($param, &$variable, int $type = ParameterType::STRING, ?int $length = null) : void
     {
-        $param = $this->_paramMap[$param];
+        if (is_int($param)) {
+            if (! isset($this->_paramMap[$param])) {
+                throw new OCI8Exception(sprintf('Could not find variable mapping with index "%d", in the SQL statement', $param));
```

Comment:
Please see https://github.com/doctrine/dbal/pull/3738#discussion_r349740890:

> Should it be `%d` instead of `"%s"` since we're printing a numeric index?